packages/mail-client/src/mail-client.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  Destination,
+  DestinationOrFetchOptions,
+  resolveDestinationWithType,
+  toDestinationNameUrl
+} from '@sap-cloud-sdk/connectivity';
+import { createLogger } from '@sap-cloud-sdk/util';
+import nodemailer, { SentMessageInfo, Transporter } from 'nodemailer';
+import { MailOptions, MailResponse } from './mail-client-types';
+
+const logger = createLogger({
+  package: 'mail-client',
+  messageContext: 'mail-client'
+});
+
+function getMailDestinationOriginalProperties(
+  destination: Destination
+): Record<string, any> {
+  const originalProperties = destination.originalProperties;
+  if (!originalProperties) {
+    throw Error(
+      `The destination '${toDestinationNameUrl(
+        destination
+      )}' does not contain property configurations.`
+    );
+  }
+  return originalProperties;
+}
+
+function getHostAndPort(
+  destination: Destination,
+  originalProperties: Record<string, any>
+): { host: string; port: number } {
+  const host = originalProperties['mail.smtp.host'];
+  const port = originalProperties['mail.smtp.port'];
+
+  if (!(host && port)) {
+    throw Error(
+      `The destination '${toDestinationNameUrl(
+        destination
+      )}' does not contain host or port information as properties.`
```

Comment:
[nth] Perhaps give a hint where this is maintend - destination property bli bla blup.